### PR TITLE
Roshan

### PR DESCRIPTION
**Pull request by AntonyRose**

### CODE REVIEW

File: azure function/README.md
Comment (JeffreySunjay4044) [inline azure function/README.md:1]: Nice habit da of maintaining the readme. Keep it up.

File: azure function/src/main/java/com/redsun/Endeca.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    private static final String DATABASE_ID = "SampleDB";
+    private static final String CONTAINER_ID = "Users";
+
+    private static CosmosClient cosmosClient = new CosmosClientBuilder()
+    .endpoint(COSMOS_CONNECTION_STRING)
+    .key(COSMOS_KEY)
+    .buildClient();
+
+
+    @FunctionName("Endeca-data")
```
Comment (JeffreySunjay4044) [inline azure function/src/main/java/com/redsun/Endeca.java]: Keep function name in lowercase

**Comment by JeffreySunjay4044** [comment]
Where are the java springboot api changes

**Comment by AntonyRose** [comment]
> Where are the java springboot api changes

@AntonyRose
[java api to fetch data](https://github.com/redsun-innovations/azure/pull/1/commits/85eaf88968cc278f1e114ba073452469168b21ae)

**Comment by JeffreySunjay4044** [comment]
package name should be in small case without special characters. no fetch_data

File: Api/src/main/java/com/redsun/fetch_data/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    public List<String> getQueryData(@RequestParam String name, @RequestParam String classCode) {
+        return userService.getQueryData(name, classCode);
+    }
+
+    @PostMapping("/list")
+    public List<String> listBase36Ids(@RequestBody List<User> facetGroups) {
+        return userService.listQueryData(facetGroups);
+    }
+
+    // GET endpoint
```
Comment (JeffreySunjay4044) [inline Api/src/main/java/com/redsun/fetch_data/controller/UserController.java]: change

File: Api/src/main/java/com/redsun/fetch_data/model/User.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.redsun.fetch_data.model;
+
+public class User {
```
Comment (JeffreySunjay4044) [inline Api/src/main/java/com/redsun/fetch_data/model/User.java]: Why User ?

File: Api/src/main/java/com/redsun/fetch_data/repository/UserRepository.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import org.springframework.stereotype.Repository;
+import com.redsun.fetch_data.model.User;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@Repository
+public class UserRepository {
```
Comment (JeffreySunjay4044) [inline Api/src/main/java/com/redsun/fetch_data/repository/UserRepository.java]: Why UserRepository ?

File: Api/src/main/java/com/redsun/fetch_data/service/UserService.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.redsun.fetch_data.model.User;
+import com.redsun.fetch_data.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class UserService {
```
Comment (JeffreySunjay4044) [inline Api/src/main/java/com/redsun/fetch_data/service/UserService.java]: Why UserService name it properly. Like facetService or facetRepository or some other name which is common to both heirachy and facet.

File: Api/src/main/java/com/redsun/fetch_data/controller/UserController.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/v1/api/facet")
+public class UserController {
+
+    private final UserService userService;
```
Comment (JeffreySunjay4044) [inline Api/src/main/java/com/redsun/fetch_data/controller/UserController.java]: Replace the constructor below with
`
@Autowired
UserService userService`

File: Api/src/main/java/com/redsun/fetch_data/service/UserService.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    private final UserRepository userRepository;
```
Comment (JeffreySunjay4044) [inline Api/src/main/java/com/redsun/fetch_data/service/UserService.java]: Same as in controller.

**Review by JeffreySunjay4044 — COMMENTED** [review]
Address the comments.

File: Api/src/main/resources/application.yaml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+server:
```
Comment (JeffreySunjay4044) [inline Api/src/main/resources/application.yaml]: Awesome da. I did not expect this.

**Pull request merged**